### PR TITLE
Add multi-currency enablement check to WooPay session handler

### LINKED ISSUE #7329: Missing Multi-Currency check could lead to errors in WooPay
### Describe the bug
I was researching other things and found a chunk of code that uses `MultiCurrency::CURRENCY_META_KEY` and `MultiCurrency::CURRENCY_SESSION_KEY` without checking if MultiCurrency is enabled or not. If MC is not enabled, it will likely cause errors.

Code is here:

https://github.com/Automattic/woocommerce-payments/blob/e5e7f69b960452cda2d7b5f9cd085718533e4152/includes/woopay/class-woopay-session.php#L348-L357

Multi-Currency is enabled by default, but is able to be disabled by a merchant in their settings. The check to determine if it is enabled is `WC_Payments_Features::is_customer_multi_currency_enabled()`.

### PR DESCRIPTION
**Pull request by c-shultz**
Fixes #7329 

#### Changes proposed in this Pull Request

Title: Add multi-currency enablement check to WooPay session handler

Description: Adding multi-currency enablement check to WooPay session handler to avoid unwanted multi-currency behavior for merchants using WooPay that have disabled multi-currency. @bborman22 describes the scenario here: https://github.com/Automattic/woocommerce-payments/issues/7329#issuecomment-1747467641

#### Testing instructions
(Instructions pulled from https://github.com/Automattic/woocommerce-payments/pull/7055#issue-1863965079)

1. Ensure WooPay is eligible and enabled.
2. Enable Multi-Currency + the currency switcher from WC > Settings > Multi-Currency.
3. Enable multiple currencies, for example: USD, JPY, EUR.
4. As a guest shopper use the currency switcher to change the currency to JPY.
5. Navigate to the checkout page and enter a valid WooPay email + OTP.
6. From the WooPay checkout, ensure the currency is displayed correctly.
7. In a separate tab, go to the merchant store and change the currency to, "USD", for example.
8. Refresh the WooPay checkout tab and ensure the currency, totals and currency symbol are set to USD.
9. Checkout with WooPay.
10. Ensure the charge is correctly processed for the chosen currency.
11. Logout from WooPay and repeat steps 4-10 as a logged in shopper.
12. Disable multi-currency and repeat steps 4-11

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7390` or branch name `fix/7329-missing-multi-currency-check` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7329-missing-multi-currency-check) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5e38077f2f0a14b6901e7c5c075726a907be6436
- Build time: 2023-10-05 19:56:50 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/woopay/class-woopay-session.php [hunk 1]
```diff
@@ -350,7 +350,7 @@ private static function get_init_session_request( $order_id = null, $key = null,
 			$customer_id   = WC_Payments::get_customer_service()->create_customer_for_user( $user, $customer_data );
 		}
 
-		if ( 0 !== $user->ID ) {
+		if ( WC_Payments_Features::is_customer_multi_currency_enabled() && ( 0 !== $user->ID ) ) {
```
Comment (bborman22) [inline includes/woopay/class-woopay-session.php]: I've always been partial to the parenthesis around conditions like `( 0 !== $user->ID )`, but for consistency we don't use them elsewhere, so we can remove them here.

**Review by bborman22 — APPROVED** [review]
I was able to test this with success. My theory on what would happen in case the user had a selected currency when multi-currency was disabled didn't play out that way as I noticed the cart data pulled the correctly currency, but it was setting the session currency without this check. So I think it's a net positive to have this additional check.

Just had a minor nit on a conditional formatting, but everything else looked and tested well for me so going to approve ahead of that small change!

LGTM!

File: includes/woopay/class-woopay-session.php (hunk 1 above)
Comment (c-shultz) [reply includes/woopay/class-woopay-session.php]: Updated! I guess we'll all just blindly trust operator precedence rules then 😀 . Thanks for reviewing!

**Pull request merged**